diff --git a/browser/app/moz.build b/browser/app/moz.build
index f41ce92556..1d2ac9ccae 100644
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -174,3 +174,20 @@ for icon in ("firefox", "document", "newwindow", "newtab", "pbmode", "document_p
         CONFIG["MOZ_BRANDING_DIRECTORY"],
         icon,
     )
+
+
+if CONFIG["OS_TARGET"] == "WINNT":
+    checkOSBuildFloorp = "windows"
+elif CONFIG["OS_TARGET"] == "Darwin":
+    checkOSBuildFloorp = "mac"
+else:
+    checkOSBuildFloorp = "linux"
+
+for icon in ["document", "document_pdf"]:
+    DEFINES[icon.upper() + "_ICO"] = '"%s/%s/%s/%s/%s.ico"' % (
+        TOPSRCDIR,
+        CONFIG["MOZ_BRANDING_DIRECTORY"],
+        "content",
+        checkOSBuildFloorp,
+        icon,
+    )
